src/types/mod.rs

```
@@ -1460,27 +1461,39 @@ pub fn execution_tree(elements: Vec<ContractTraceElement>) -> Option<ExecutionTr
+                        .map(|version| match version {
+                            WasmVersion::V0 => ExecutionTree::V0(ExecutionTreeV0 {
+                                top_level: UpdateV0 {
+                                    address,
+                                    instigator,
+                                    amount,
+                                    message,
+                                    receive_name,
+                                    events,
+                                },
+                                rest:      Vec::new(),
+                            }),
+
+                            WasmVersion::V1 => ExecutionTree::V1(ExecutionTreeV1 {
                                 address,
                                 instigator,
                                 amount,
                                 message,
                                 receive_name,
-                                events,
-                            },
-                            rest:      Vec::new(),
-                        }),
-                        WasmVersion::V1 => ExecutionTree::V1(ExecutionTreeV1 {
-                            address,
-                            instigator,
-                            amount,
-                            message,
-                            receive_name,
-                            events: vec![TraceV1::Events { events }],
-                        }),
-                    };
+                                events: vec![TraceV1::Events { events }],
+                            }),
+                        })
+                        .map_err(|err| {
+                            tonic::Status::internal(format!(
+                                "Issue mapping to WasmVersion for WasmVersionInt: \
+                                 {contract_version} error: {err}"
+                            ))
+                        })
+                        .ok()?;
```

This line "swallows" the error you just created and returns `None`. That does not seems correct, I believe you should change the method signature to be a result and return the error.